app/app.js

```diff
@@ -0,0 +1,68 @@
+var express = require('express');
+var path = require('path');
+var logger = require('morgan');
+var session = require('express-session');
+var passport = require('passport');
+var Auth0Strategy = require('passport-auth0-openidconnect').Strategy;
+
+var config = require('./config');
+var routes = require('./routes');
+
+// This will configure Passport to use Auth0
+const strategy = new Auth0Strategy(
+  config.auth0,
+  ((req, issuer, audience, profile, accessToken, refreshToken, params, callback) => {
```

Comment:
As the request is not used in this function, you should probably change the `passReqToCallback` setting to `false` in `config.auth0` and use the simpler function signature:
```javascript
(issuer, audience, profile, callback) => {
```

Reply:
I think that was the weird problem we had if you remember.
Even when not setting anything in the session explicitly here that `passReqToCallback` seems to be necessary - I'll give it another quick try but we tried last time and it wan't working without that set to `true`.

My theory was that if request is not passed back will not have cookie and therefore will not create user session - I don't know express enough to tell to be honest.

Reply:
I may be wrong but I think really depends on the strategy library used as I've notice the other one used in another project had a different callback signature.

Not sure if just rename and cut some of the params is safe here. I'll probably err on the safe side and leave the signature alone.

Reply:
FYI/To the future me: Tried again and yes, `passReqToCallback` needs to be `true` or it will not authenticate the user.

Reply:
That function signature is from the [`passport-auth0-openidconnect` README](https://github.com/auth0/passport-auth0-openidconnect/blob/master/README.md).

If you set `passReqToCallback = false` you have to also change the function signature, it won't work if you do one or the other.

Reply:
OK, I'll give another shot to the this combination `passReqToCallback = false` and different signature and see if that works 👍

Reply:
It worked :) https://github.com/ministryofjustice/analytics-platform-kibana-auth-proxy/pull/1/commits/287e52ab54cf943f1801e949bd01dcc78e890fb5